cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowStatsSLAHttpHandlerTest.java

```diff
@@ -283,23 +290,32 @@ public void testCompare() throws Exception {
     List<RunId> runIdList = new ArrayList<>();
     long startTime = System.currentTimeMillis();
     long currentTimeMillis;
+
+    // Time in seconds from program starting to program running
+    int startDelaySecs = 5;
+
     for (int i = 0; i < count; i++) {
       // work-flow runs every 5 minutes
       currentTimeMillis = startTime + (i * TimeUnit.MINUTES.toMillis(5));
       RunId workflowRunId = RunIds.generate(currentTimeMillis);
       runIdList.add(workflowRunId);
-      store.setStart(workflowProgram, workflowRunId.getId(), RunIds.getTime(workflowRunId, TimeUnit.SECONDS));
+      long workflowStartTimeSecs = TimeUnit.MILLISECONDS.toSeconds(currentTimeMillis);
+      store.setStartAndRun(workflowProgram, workflowRunId.getId(), workflowStartTimeSecs,
+                           workflowStartTimeSecs + startDelaySecs);
 
       // MR job starts 2 seconds after workflow started
-      RunId mapreduceRunid = RunIds.generate(currentTimeMillis + TimeUnit.SECONDS.toMillis(2));
+      RunId mapreduceRunid = RunIds.generate(currentTimeMillis
+                                             + TimeUnit.SECONDS.toMillis(2)
+                                             - TimeUnit.SECONDS.toMillis(startDelaySecs));
```

Comment:
Needed to make sure runs are considered in the "running" state as the same time as before the change.